File: utils/getTeamColor.ts
const getTeamColor = (
  year: string,
  round: string,
  constructor_name: string,
) => {
  return fetch(
    `${process.env.NEXT_PUBLIC_API_URL}py/team-color/?year=${year}&round_number=${round}&constructor_name=${constructor_name}`,
  )
    .then((response) => {
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      return response.json()
    })
    .then((data) => {
      return data
    })
}

export default getTeamColor

File: api/utils/get_team_color.py
import fastf1

def get_team_color(year: int, round: int, constructor_name: str):
    try:
        session = fastf1.get_session(year, round, 5)

        team_color = fastf1.plotting.get_team_color(constructor_name, session)
       
        return team_color
    except Exception as e:
        return {"error": str(e)}